src/mapper/pkg/pod_webhook/pod_webhook.go

```diff
@@ -0,0 +1,120 @@
+package pod_webhook
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/otterize/network-mapper/src/shared/kubeutils"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"net/http"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+type InjectDNSConfigToPodWebhook struct {
+	client           client.Client
+	decoder          *admission.Decoder
+	currentNamespace string
+	dnsServerAddress string
+}
+
+func NewInjectDNSConfigToPodWebhook(client client.Client, decoder *admission.Decoder) *InjectDNSConfigToPodWebhook {
+	podNamespace, err := kubeutils.GetCurrentNamespace()
+
+	if err != nil {
+		logrus.WithError(err).Panic("unable to get pod namespace")
+	}
+
+	dnsServerAddress, err := getDNSServerAddress(client, podNamespace)
+
+	if err != nil {
+		logrus.WithError(err).Panic("unable to get DNS server address")
```

Comment:
Don't panic here (in a constructor - `New`) - panic is only allowed in main for error handling